### PR TITLE
Webpack config using JavaScript preprocessor syntax

### PR DESCRIPTION
**Pull request by ezze**
This pull request allows to use [ES6 syntax in webpack config file](https://github.com/zinserjan/mocha-webpack/issues/51).

### CODE REVIEW

**Comment by ezze** [comment]
Travis CI test failed for Node 0.12 and Wepback beta 2.1.0-beta.25 due to [this issue](https://github.com/webpack/webpack/issues/3041).

It's fixed by [this pull request](https://github.com/webpack/webpack/pull/3080) and, I guess, must be in next Webpack beta release.

**Comment by zinserjan** [comment]
Thanks for your PR.

I think when we want to support this, we should do this the right way and make this more generic. This works only for babel.

I would like to have a solution which works with all other preprocessors.

Also we should handle the preprocessors  extensions automatically. For example if we have a *webpack.config-test.babel.js* file on disk the following command should also work
```bash
$ mocha-webpack --webpack-config webpack.config-test.js
```

Webpack uses [interpret](https://www.npmjs.com/package/interpret) for this. See [here](https://github.com/webpack/webpack/blob/9c5c6df64f7c53f014be058fa18afde03fe6b77f/bin/convert-argv.js#L40-L127).

Could you have a look at this?

**Comment by ezze** [comment]
Thanks for the reply.

I'm relatively new to Webpack and have dealt with Babel preprocessor only but after looking at the sources you linked to, I guess, it doesn't seem too difficult to implement.

I'll try to delve into this later (probably, in a few days) when I have more time.

**Comment by ezze** [comment]
@zinserjan, please, check this.

I've added simple implementation of preprocessors support based on [interpret](https://www.npmjs.com/package/interpret) and Webpack implementation.

Also tests for plain JavaScript, Babel and CoffeeScript webpack configuration files are added.

File: README.md [hunk 1]
```diff
@@ -81,6 +81,28 @@ Maybe you noticed that [entry](https://webpack.github.io/docs/configuration.html
...
+```
+
+**webpack.config-test.coffee** - CoffeeScript example config
+```coffee
+nodeExternals = require 'webpack-node-externals'
+
+module.exports =
+  target: 'node'
+  externals: [nodeExternals()]
+```
```
Comment (zinserjan) [inline README.md]: Can you also add here a simple coffee loader configuration, e.g.

```js
 module:
   loaders: [
            { test: /\.coffee$/, loader: "coffee-loader" }
        ]
```

Comment (ezze) [inline README.md]: Added for both Babel and Coffee.

Comment (zinserjan) [inline README.md]: Nice! Thanks

**Pull request merged**